## Задача 2

Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:   
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.  

Известно, что в генеральной совокупности IQ распределен нормально.   
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Для нахождения доверительного интервала для математического ожидания нам нужно знать среднее значение выборки и её стандартное отклонение.

Используем стандартное отклонение выборки в качестве оценки стандартного отклонения генеральной совокупности, так как мы не знаем его истинного значения:

In [1]:
import numpy as np

sample = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
n = len(sample)
mean = np.mean(sample)
std_dev = np.std(sample, ddof=1)

Здесь n - размер выборки, mean - среднее значение выборки, std_dev - исправленное стандартное отклонение выборки.

Теперь мы можем найти доверительный интервал для математического ожидания при помощи формулы:

In [2]:
import scipy.stats as stats

confidence_level = 0.95
alpha = 1 - confidence_level
t = stats.t.ppf(1 - alpha / 2, df=n-1)  # критическое значение t-статистики

lower = mean - t * std_dev / np.sqrt(n)
upper = mean + t * std_dev / np.sqrt(n)

print("Доверительный интервал: ({:.2f}, {:.2f})".format(lower, upper))

Доверительный интервал: (110.56, 125.64)


Здесь t - критическое значение t-статистики, которое мы находим с помощью функции stats.t.ppf() из библиотеки scipy.stats.

Итак, доверительный интервал для математического ожидания с надежностью 0.95 для данной выборки равен (110.56, 125.64). То есть мы можем утверждать, что с 95% уверенностью истинное значение математического ожидания находится в этом интервале.